src/Oro/Bundle/DistributionBundle/OroKernel.php

```diff
@@ -152,15 +152,35 @@ protected function collectBundles()
 
         uasort($bundles, array($this, 'compareBundles'));
 
+        do {
+            $bundleNames = array_keys($bundles);
+            $removedBundles = array_filter($bundles, function ($config) use ($bundleNames) {
+                if (!isset($config['dependencies'])) {
+                    return false;
+                }
+
```

Comment:
@gplanchat  This doesn't look efficient as we are filtering out the same bundles without dependencies again and again in each loop.

Reply:
Yes, thus there is a good reason for this : disabling a bundle can remove a dependency of another bundle and require it to be disabled on 2nd pass, etc.

This processing is made once, at cache warmup, it don't affect runtime

Reply:
@gplanchat This solution has some big drawback, we can't lock required version of the package with the functionality `suggestion`. In case if suggested bundle changes own API we will receive broken functionality, I think you agree with me that this inappropriate for the system like OroPlatform.

I think better solution when we host bridge bundle separately from the monolithic repository and include it only when we need it. This bundle will have strong dependencies, so we know that if someone wants to use this functionality he receive worked solution.

Reply:
@gplanchat  I understand why it is done, I'm just saying that it is not an optimal (algorithmically) way to do it.

Reply:
@alexandr-parkhomenko I don't think there is a reasonable way to integrate composer in this case. The goal is to keep things simple.

3 things can help to maintain compatibility :
* Oro packages require specific OroPlatform version, which is a common dependency managed by Composer, your solution about declared dependencies in the `composer.json`of the bridge bundle actually does the version compatibility check and is not exclusive with this option
* SemVer guarantees to not break things at any time, if it is properly followed
* a `README.md` file with a version compatibility list could solve most cases

There are not thousands of Oro packages for now, most of the needs you are mentioning can be handled by composer (which is the best option). This change provide a way to tell Oro to not load the bundle in the kernel when all dependencies are not met.

@mbessolov maybe we could Blackfire it to find some better algorithm

Reply:
@alexandr-parkhomenko let me describe you a broken use case happening on current versions:

In `FooBundle`
```yaml
bundles:
  - name: Foo\FooBundle
``` 

In `BarBundle`
```yaml
bundles:
  - name: Bar\BarBundle
``` 

In `FooBarBridgeBundle`
```yaml
bundles:
  - name: Foo\FooBarBridgeBundle
``` 

In `BazBundle`
```yaml
bundles:
  - name: Baz\BazBundle
exclusions:
  - name: Bar\BarBundle
``` 

The `BarBundle` will be excluded by the `BazBundle`, but not the bridge which will be loaded either by composer and the OroKernel.